crates/symbolicator-service/tests/integration/sourcemap.rs

```diff
@@ -0,0 +1,300 @@
+    assert_eq!(frames[0].raw.lineno, Some(6));
+    assert_eq!(frames[0].raw.filename, Some("index.html".to_string()));
+
+    assert_eq!(frames[1].raw.function, Some("test".to_string()));
+    assert_eq!(frames[1].raw.lineno, Some(20));
+    assert_eq!(frames[1].raw.filename, Some("test.js".to_string()));
+
+    assert_eq!(frames[2].raw.function, Some("invoke".to_string()));
+    assert_eq!(frames[2].raw.lineno, Some(15));
+    assert_eq!(frames[2].raw.filename, Some("test.js".to_string()));
+
+    assert_eq!(frames[3].raw.function, Some("onFailure".to_string()));
+    assert_eq!(frames[3].raw.lineno, Some(5));
+    assert_eq!(frames[3].raw.filename, Some("test.js".to_string()));
+}
+
+// TODO(kamil): Failing due to source context not resolving file content from non-sourceContents locations.
+#[ignore]
+#[tokio::test]
+async fn test_sourcemap_source_expansion() {
+    let (symbolication, _) = setup_service(|_| ());
+    let srv = symbolicator_test::sourcemap_server("02_sourcemap_source_expansion");
+
+    let stacktraces: Vec<JsProcessingRawStacktrace> = serde_json::from_str(
+        r#"[{
+            "frames": [
+                {
+                    "function": "function: \"HTMLDocument.<anonymous>\"",
+                    "abs_path": "http//example.com/index.html",
+                    "filename": "index.html",
+                    "lineno": 283,
+                    "colno": 17,
+                    "in_app": false
+                },
+                {
+                    "abs_path": "http://example.com/file.min.js",
+                    "filename": "file.min.js",
+                    "lineno": 1,
+                    "colno": 39
+                }
+            ]
+        }]"#,
+    )
+    .unwrap();
+
+    let request = JsProcessingSymbolicateStacktraces {
+        source: Arc::new(SentrySourceConfig {
+            id: SourceId::new("sentry:project"),
+            url: srv.url("/files/"),
+            token: String::new(),
+        }),
+        stacktraces,
+        dist: None,
+    };
+
+    let response = symbolication.js_processing_symbolicate(request).await;
+    let frames = &mut response.unwrap().stacktraces[0].frames;
+
+    assert_eq!(frames.len(), 2);
+
+    assert_eq!(
+        frames[1].raw.pre_context,
+        &["function add(a, b) {", "\t\"use strict\";"]
+    );
+    assert_eq!(
+        frames[1].raw.context_line,
+        Some("\treturn a + b; // fôo".to_string())
+    );
+    assert_eq!(frames[1].raw.post_context, &["}", ""]);
+}
+
+#[tokio::test]
+async fn test_sourcemap_embedded_source_expansion() {
+    let (symbolication, _) = setup_service(|_| ());
+    let srv = symbolicator_test::sourcemap_server("05_sourcemap_embedded_source_expansion");
+
+    let stacktraces: Vec<JsProcessingRawStacktrace> = serde_json::from_str(
+        r#"[{
+            "frames": [
+                {
+                    "function": "function: \"HTMLDocument.<anonymous>\"",
```

Comment:
I don’t think thats intentional? :-D

Reply:
It is. It will test the error case later on, that's why I copy the payloads literally from monolith :p